client/src/partials/subsidy/subsidy.js

```diff
@@ -0,0 +1,103 @@
+angular.module('bhima.controllers')
+.controller('subsidy', [
+  '$scope',
+  '$translate',
+  '$http',
+  'validate',
+  'messenger',
+  'connect',
+  'appstate',
+  'uuid',
+  function ($scope, $translate, $http, validate, messenger, connect, appstate, uuid) {
+    var dependencies = {},
+        session = $scope.session = {};
+
+    dependencies.subsidies = {
+      identifier:'uuid',
+      query : '/getSubsidies/'
+    };
+
+    dependencies.debitorGroups = {
+      query : {
+        identifier : 'uuid',
+        tables : {
+          'debitor_group' : {columns : ['uuid', 'name']}
+        }
+      }
+    };
+
+    dependencies.enterprise = {
+      query : {
+        tables : {
+          'enterprise' : { columns : ['id'] },
+          'currency' : {columns : ['symbol']}
+        },
+        join : ['enterprise.currency_id=currency.id']
+      }
+    };
+
+    function startup (models) {
+      angular.extend($scope, models);
+    }
+
+    validate.process(dependencies).then(startup);
+
+    $scope.delete = function (subsidy) {
+      var result = confirm($translate.instant('SUBSIDY.CONFIRM'));
+      if (result) {
+        connect.delete('subsidy', 'uuid', [subsidy.uuid])
+        .then(function () {
+          $scope.subsidies.remove(subsidy.uuid);
+          messenger.info($translate.instant('SUBSIDY.DELETE_SUCCESS'));
+        });
+      }
+    };
+
+    $scope.edit = function (subsidy) {
+      console.log('le voici', subsidy);
+      session.action = 'edit';
+      session.edit = angular.copy(subsidy);
+    };
+
+    $scope.new = function () {
+      session.action = 'new';
+      session.new = {};
+    };
+
+    $scope.save = {};
+
+    $scope.save.edit = function () {
+
+      var record = angular.copy(connect.clean(session.edit));
```

Comment:
connect.clean() will do essentially the same thing as angular.copy().  You can remove the angular.copy() call.